src/wasm/extract.hpp

```diff
@@ -20,6 +20,8 @@
 
 #include "wasm/fdzipstream/fdzipstream.h"
 
+#define wasm_debug(X) do { if (extractor::get().get_settings().debug_messages_enabled) log_info << X; } while(0)
```

Comment:
Do we need that construction here?
`do { ... } while(0)`

Reply:
It's a popular way to create a macro wrapper that doesn't produce any errors, no matter how it's used in the code.
wasm_info is just wrapped here with an additional conditional statement, making this output the log only if the debug output is enabled before starting the extraction.